core/amber/src/main/scala/edu/uci/ics/texera/web/resource/GoogleResource.java

```diff
@@ -0,0 +1,83 @@
+import com.google.api.client.googleapis.auth.oauth2.GoogleClientSecrets;
+import com.google.api.client.googleapis.javanet.GoogleNetHttpTransport;
+import com.google.api.client.http.javanet.NetHttpTransport;
+import com.google.api.client.json.JsonFactory;
+import com.google.api.client.json.jackson2.JacksonFactory;
+import com.google.api.client.util.store.FileDataStoreFactory;
+import com.google.api.services.drive.Drive;
+import com.google.api.services.drive.DriveScopes;
+import com.google.api.services.sheets.v4.Sheets;
+import com.google.api.services.sheets.v4.SheetsScopes;
+import edu.uci.ics.texera.web.WebUtils;
+
+import java.io.*;
+import java.security.GeneralSecurityException;
+import java.util.Arrays;
+import java.util.List;
+
+public class GoogleResource {
+    private static final String APPLICATION_NAME = "Texera";
+    private static final String TOKENS_DIRECTORY_PATH = WebUtils.getTokenPath();
+    private static final JsonFactory JSON_FACTORY = JacksonFactory.getDefaultInstance();
+
+    /**
+     * Global instance of the scopes required by this quickstart.
+     * If modifying these scopes, delete your previously saved tokens/ folder.
+     */
+    private static final String CREDENTIALS_FILE_PATH = WebUtils.getCredentialPath();
+    private static final List<String> SCOPES = Arrays.asList(SheetsScopes.SPREADSHEETS, DriveScopes.DRIVE);
+
+    public static Sheets createSheetService() throws IOException {
+        try {
+            NetHttpTransport httpTransport = GoogleNetHttpTransport.newTrustedTransport();
+            Credential credential = getCredentials(httpTransport);
+            return new Sheets.Builder(httpTransport, JSON_FACTORY, credential)
+                    .setApplicationName(APPLICATION_NAME)
+                    .build();
+        } catch (GeneralSecurityException e) {
+            e.printStackTrace();
+            return null;
+        }
+    }
+
+    public static Drive createDriveService() throws IOException {
+        try {
+            NetHttpTransport httpTransport = GoogleNetHttpTransport.newTrustedTransport();
+            Credential credential = getCredentials(httpTransport);
+            return new Drive.Builder(httpTransport, JSON_FACTORY, credential)
+                    .setApplicationName(APPLICATION_NAME)
+                    .build();
+        } catch (GeneralSecurityException e) {
+            e.printStackTrace();
+            return null;
+        }
+    }
+
+    /**
+     * Creates an authorized Credential object for the service account
+     */
+    private static Credential getCredentials(NetHttpTransport httpTransport) throws IOException {
+        // Load client secrets test
+        File initialFile = new File(CREDENTIALS_FILE_PATH);
+        InputStream targetStream = new FileInputStream(initialFile);
+        if (targetStream == null) {
+            throw new FileNotFoundException("Resource not found: " + CREDENTIALS_FILE_PATH);
```

Comment:
denote this exception to the method signature

Reply:
`IOException` is the super class for `FileNotFoundException`.  `IOException` is already in method signature so I believe there's no need to mention again.